mt-kahypar/partition/context.cpp

```diff
@@ -205,19 +211,31 @@ namespace mt_kahypar {
   }
 
   void Context::setupPartWeights(const HypernodeWeight total_hypergraph_weight) {
-    partition.perfect_balance_part_weights.clear();
-    partition.perfect_balance_part_weights.push_back(ceil(
-            total_hypergraph_weight
-            / static_cast<double>(partition.k)));
-    for (PartitionID part = 1; part != partition.k; ++part) {
-      partition.perfect_balance_part_weights.push_back(
-              partition.perfect_balance_part_weights[0]);
-    }
-    partition.max_part_weights.clear();
-    partition.max_part_weights.push_back((1 + partition.epsilon)
-                                         * partition.perfect_balance_part_weights[0]);
-    for (PartitionID part = 1; part != partition.k; ++part) {
-      partition.max_part_weights.push_back(partition.max_part_weights[0]);
+    if (partition.use_individual_part_weights) {
+      ASSERT(static_cast<size_t>(partition.k) == partition.max_part_weights.size());
```

Comment:
is not necessary to also calculate an epsilon here based on the weight_fraction? Some algorithms require that, especially the global rollback algorithm that Lars mentioned.

Reply:
Whoops :sweat_smile: